### ISSUE TITLE
Encrypting with no padding inflates cipher text

### ISSUE THREAD

**Issue by paulw11** [TakeScoop/SwiftyRSA#29]
See [SecKeyEncrypt SecPaddingNone iOS](http://stackoverflow.com/questions/39114114/seckeyencrypt-secpaddingnone-ios)

> I am basically doing encryption with two keys. The first step, using key A, uses SecPaddingPKCS1 and gives me an output of 256 bytes. After that, using Key B, I use SecPaddingNone. I assume that my output will remain 256 bytes, but it becomes 512 bytes. I am not sure what I am doing wrong, does anyone know? Am I thinking about this wrong?

**Comment by paulw11** [comment]
RSA returns blocksize (in this case 128) bytes after encryption. If you encrypt more than blocksize bytes then the data is split into multiple chunks, each of which is encrypted, so you will get a multiple of blocksize bytes back.

When PKCS1 padding is used the effective blocksize is reduced by 11 bytes, which means that encrypting 128 bytes will return two blocks or 256 bytes (since 128 is > 128-11 or 117).

When no padding is used, the blocksize doesn't need to be reduced by 11, but SwiftyRSA still does this. I have tested without the 11 byte reduction using no padding and the second encryption, without padding, returns 128 bytes as expected. The regression tests still pass and I have confirmed that openssl can correctly decrypt the double encrypted data (first with padding and then encrypted a second time with no padding).
